### DISCUSSION TITLE
Using two copies of AM in parallel

### DISCUSSION THREAD

**Discussion by aleksejrs** [mortii/anki-morphs#176] [General]
I made a copy of AM and replaced all instances of "ankimorph", "AnkiMorph" and "ANKIMORPH" with "ankim2amorphs", "AnkiMorph" and "ANKIMORPH", and also "am_tool_menu" with "amm2a_tool_menu". The new AM added a menu item, but still the original AM received the settings changed by the new AM. Why is it so hard?
I wanted to track the morphs in movies2anki cards separately, because it's just the ability to transcribe and I've already transcribed a word I don't know the meaning of.

**Comment by mortii** [comment]
In the anki profile folder there is a file called `ankimorphs_profile_settings.json`. This file is loaded when an anki profile is opened, so that is likely your problem. You probably want to rename that file to make AM not read it, or you might want to use a different profile for your altered AM version, that way the file won't interfere with the original AM settings and vice versa.  

I'll close this discussion now, but you can keep using it if you have more questions. 

EDIT:
or you could comment out this line:
https://github.com/mortii/anki-morphs/blob/e8c3634eb685baefa8383a89cbdf6304f5f04774/ankimorphs/__init__.py#L71

**Reply by aleksejrs** [reply]
But I replaced "ankimorphs" in that file's name, too:

ankim2amorphs_globals.py
```python3
PROFILE_SETTINGS_FILE_NAME = "ankim2amorphs_profile_settings.json"
```

Is "load_am_profile_configs" shared?

**Comment by mortii** [comment]
Add-ons basically only have one settings file (shared if you will) which is `addons21/[addon]/meta.json`. 

So the `ankimorphs_profile_settings.json` is just used to update that shared `meta.json` file. You can see it in action here: 

https://github.com/mortii/anki-morphs/blob/6e3d3be6eec295e457d91cc4a05326880a31dabe/ankimorphs/ankimorphs_config.py#L202-L216

`mw.addonManager.getConfig(__name__)` extracts that meta.json dictionary

`mw.addonManager.writeConfig(__name__, config)` updates meta.json

The fact that `meta.json` is shared across profiles is what makes this process so convoluted; Anki basically does not support add-ons having per profile settings. 

So I assume it's the `__name__` part that is causing problems then, maybe you didn't fully refactor the name of your custom module.
